internal/checker/utilities.go

```diff
@@ -1218,8 +1218,33 @@ func isValidNumberString(s string, roundTripOnly bool) bool {
 	return !n.IsNaN() && !n.IsInf() && (!roundTripOnly || n.String() == s)
 }
 
-func isValidBigIntString(s string, roundTripOnly bool) bool {
-	return false // !!!
+func isValidBigIntString(s string, roundtrip bool) bool {
+	if s == "" {
+		return false
+	}
+	scan := scanner.NewScanner()
+	text := s + "n"
+	success := true
+	scan.SetOnError(func(diagnostic *diagnostics.Message, start, length int, args ...any) {
+		success = false
+	})
+	scan.SetText(text)
+	result := scan.Scan()
+	negative := result == ast.KindMinusToken
+	if negative {
+		result = scan.Scan()
+	}
+	if !success || result != ast.KindBigIntLiteral || scan.TokenEnd() != len(text) || scan.TokenFlags()&ast.TokenFlagsNumericLiteralFlags != 0 {
```

Comment:
This logic seems a little different than the original code. I know you tried to flatten it, but the TokenEnd thing is off by one.

Reply:
updated with I think is the right version now